### PR TITLE
 loggly detector

### PR DESCRIPTION
**Pull request by ankushgoel27**
detects the solarwinds loggly api token along with the custom domain

How to generate the token.
https://documentation.solarwinds.com/en/success_center/loggly/content/admin/token-based-api-authentication.htm

### CODE REVIEW

File: pkg/detectors/loggly/loggly.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        domainMatches := domainPat.FindAllStringSubmatch(dataStr, -1)
+
+        for _, match := range keyMatches {
+                if len(match) != 2 {
+                        continue
+                }
+                key := strings.TrimSpace(match[1])
+
+                for _, domainMatch := range domainMatches {
+                        if len(domainMatch) != 2 {
```
Comment (zricethezav) [inline pkg/detectors/loggly/loggly.go:47]: @ankushgoel27 tests need to be updated to reflect the capture of 2 matches. Right now the tests are only supplying a sample key but leave out the domain.

**Comment by zricethezav** [comment]
Hi @ankushgoel27, thanks for submitting this PR. Did you verify the tests were working locally? I.e, did you use a sample loggly account to test before opening this PR?

File: pkg/detectors/loggly/loggly.go (hunk 1 above)
Comment (ankushgoel27) [reply pkg/detectors/loggly/loggly.go:47]: Sorry, Dont understand what you mean.

**Comment by ankushgoel27** [comment]
Hi,

Yes i tested this PR with sample creds

File: pkg/detectors/loggly/loggly_test.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+		want                []detectors.Result
+		wantErr             bool
+		wantVerificationErr bool
+	}{
+		{
+			name: "found, verified",
+			s:    Scanner{},
+			args: args{
+				ctx:    context.Background(),
+				data:   []byte(fmt.Sprintf("You can find a loggly secret %s within", secret)),
```
Comment (zricethezav) [inline pkg/detectors/loggly/loggly_test.go:48]: @ankushgoel27 https://github.com/trufflesecurity/trufflehog/pull/1782/files#r1336331681 the `data` only includes a `secret`. This detector has multi-part credentials and requires both a domain and secret to be verified. Does that make sense? Make sure you have verified _this_ test is passing please

File: pkg/detectors/loggly/loggly.go (hunk 1 above)
Comment (ankushgoel27) [reply pkg/detectors/loggly/loggly.go:47]: fixed the loggly_test.go

File: pkg/detectors/loggly/loggly_test.go (hunk 2 above)
Comment (ankushgoel27) [reply pkg/detectors/loggly/loggly_test.go]: fixed the loggly_test.go

**Comment by ankushgoel27** [comment]
Hi,

Is there anything else left regarding this detector?


Thanks,

**Comment by ankushgoel27** [comment]
> @ankushgoel27 Just some merge conflicts that need to be resolved :)

Those conflicts were not there when i opened the PR

**Comment by rosecodym** [comment]
> How do i fix the hex conflict in the detectors.pb.go?

That file is automatically generated, so fortunately, you don't have to. When you're resolving conflicts, just resolve conflicts in that file any way at all. It doesn't matter what you do, because you'll be overwriting the change immediately. Once you've resolved all the conflicts everywhere, rerun `make protos` like you did originally - this will overwrite `detectors.pb.go` with a clean, correct version.

**Comment by ankushgoel27** [comment]
all merge conflicts resolved. Please merge this commit

**Comment by zricethezav** [comment]
@ankushgoel27 in order to remain fair to the competition can you provide two pieces of information as described in the guidelines:
1. a screenshot of the loggly detector test passing
2. documentation on how to generate an API token

**Comment by ankushgoel27** [comment]
I am seeing these errors. Any idea how to resolve these?

--- FAIL: TestLoggly_FromChunk (3.44s)
    --- FAIL: TestLoggly_FromChunk/found,_would_be_verified_if_not_for_timeout (0.00s)
        loggly_test.go:142: Loggly.FromData() found, would be verified if not for timeout diff: (-got +want)
              []detectors.Result(
            -   nil,
            +   {{DetectorType: s"Loggly"}},
              )
    --- FAIL: TestLoggly_FromChunk/found,_verified_but_unexpected_api_surface (0.00s)
        loggly_test.go:142: Loggly.FromData() found, verified but unexpected api surface diff: (-got +want)
              []detectors.Result(
            -   nil,
            +   {{DetectorType: s"Loggly"}},
              )
FAIL
FAIL    github.com/trufflesecurity/trufflehog/v3/pkg/detectors/loggly   3.454s
FAIL

**Comment by ankushgoel27** [comment]
loggly

All tests are passing

**Comment by ankushgoel27** [comment]
@zricethezav can we please merge this before it again shows merge conflicts??

**Review by ahrav — APPROVED** [review]

**Pull request merged**